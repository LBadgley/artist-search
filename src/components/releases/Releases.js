import React from 'react';
import Release from './Release';
import PropTypes from 'prop-types';

function Releases({ releases, artist }) {
  const releaseList = releases.map(release => {
    return (
      <li key={release.id}>
        <Release releaseName={release.title} releaseArt={release.coverArt} releaseId={release.id} artist={artist} />
      </li>
    );
  });
  
  return (
    <ul>{releaseList}</ul>
  );
}
    
Releases.propTypes = {
  releases: PropTypes.array.isRequired,
  artist: PropTypes.string
};

export default Releases;
