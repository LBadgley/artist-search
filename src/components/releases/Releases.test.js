import React from 'react';
import { shallow } from 'enzyme';
import Releases from './Releases';

describe('Releases test', () => {
  it('renders the Releases', () => {
    const releases = [
      {
        releaseName: '2342',
        releaseArt: '333',
        id: 'dsfds',
        artist: 'abcdf'
      }
    ];
    const wrapper = shallow(<Releases releases={releases} />);
    expect(wrapper).toMatchSnapshot();
  });
});
