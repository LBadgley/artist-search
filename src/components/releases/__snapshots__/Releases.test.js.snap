// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Releases test renders the Releases 1`] = `
<ul>
  <li
    key="dsfds"
  >
    <Release
      releaseId="dsfds"
    />
  </li>
</ul>
`;
